packages/admin/src/pages/super-admin-dashboard/index.page.tsx

```diff
@@ -1,40 +1,80 @@
 import { GetServerSidePropsContext } from 'next';
 import Link from 'next/link';
-import { Button, Checkboxes, Table } from 'gap-web-ui';
+import {
+  Button,
+  Checkboxes,
+  FlexibleQuestionPageLayout,
+  QuestionPageGetServerSideProps,
+  Table,
+} from 'gap-web-ui';
 import Meta from '../../components/layout/Meta';
 import PaginationType from '../../types/Pagination';
 import { getUserTokenFromCookies } from '../../utils/session';
 import { Pagination } from '../../components/pagination/Pagination';
 import styles from './superadmin-dashboard.module.scss';
-import { getSuperAdminDashboard } from '../../services/SuperAdminService';
-import { User } from './types';
+import {
+  filterUsers,
+  getSuperAdminDashboard,
+} from '../../services/SuperAdminService';
+import {
+  SuperAdminDashboardFilterData,
+  SuperAdminDashboardResponse,
+  User,
+} from './types';
 import Navigation from './Nagivation';
 import InferProps from '../../types/InferProps';
+import getConfig from 'next/config';
 
-export const getServerSideProps = async ({
-  req,
-  query,
-}: GetServerSidePropsContext) => {
+const formatRequestBody = (body: SuperAdminDashboardFilterData) => {
+  if (typeof body.roles === 'string') body.roles = [body.roles];
+  if (typeof body.departments === 'string')
+    body.departments = [body.departments];
+  if (!body.departments) body.departments = [];
+  if (!body.roles) body.roles = [];
+
+  const clearAllFilters = body['clear-all-filters'] === '' ? true : false;
+  return {
+    ...body,
+    clearAllFilters,
+  };
+};
+
+export const getServerSideProps = async (
+  context: GetServerSidePropsContext
+) => {
   const paginationParams: PaginationType = {
     paginate: true,
-    page: Number(query.page || 1) - 1,
-    size: Number(query.limit || 10),
+    page: Number(context.query.page || 1) - 1,
+    size: Number(context.query.limit || 10),
   };
 
-  const userToken = getUserTokenFromCookies(req);
-  const { departments, roles, users, userCount } = await getSuperAdminDashboard(
-    paginationParams,
-    userToken
-  );
+  const userToken = getUserTokenFromCookies(context.req);
 
-  return {
-    props: {
-      departments,
-      roles,
-      users,
-      userCount,
-    },
+  const fetchPageData = async () =>
+    await getSuperAdminDashboard(paginationParams, userToken);
+
+  const handleRequest = async (body: SuperAdminDashboardFilterData) => {
+    const res = await filterUsers(
+      paginationParams,
+      formatRequestBody(body),
+      userToken
+    );
+    return res;
   };
```

Comment:
can use implicit return here

```suggestion
  const handleRequest = async (body: SuperAdminDashboardFilterData) => 
     filterUsers(
      paginationParams,
      formatRequestBody(body),
      userToken
    );
```